satpy/readers/mwr_l1b.py

```diff
@@ -0,0 +1,201 @@
+                            "solar_azimuth_horn1",
+                            "solar_azimuth_horn2",
+                            "solar_azimuth_horn3",
+                            "solar_azimuth_horn4",
+                            "solar_zenith_horn1",
+                            "solar_zenith_horn2",
+                            "solar_zenith_horn3",
+                            "solar_zenith_horn4",
+                            "satellite_azimuth_horn1",
+                            "satellite_azimuth_horn2",
+                            "satellite_azimuth_horn3",
+                            "satellite_azimuth_horn4"]
+
+class AWS_EPS_Sterna_BaseFileHandler(NetCDF4FileHandler):
+    """Base class implementing the AWS/EPS-Sterna MWR Level-1b&c Filehandlers."""
+
+    def __init__(self, filename, filename_info, filetype_info, auto_maskandscale=True):
+        """Initialize the handler."""
+        super().__init__(filename, filename_info, filetype_info,
+                         cache_var_size=10000,
+                         cache_handle=True)
+        self.filename_info = filename_info
+
+    @property
+    def start_time(self):
+        """Get the start time."""
+        return self.filename_info["start_time"]
+
+    @property
+    def end_time(self):
+        """Get the end time."""
+        return self.filename_info["end_time"]
+
+    @property
+    def sensor(self):
+        """Get the sensor name."""
+        # This should have been self["/attr/instrument"]
+        # But the sensor name is currently incorrect in the ESA level-1b files
+        return "mwr"
+
+    @property
+    def platform_name(self):
+        """Get the platform name."""
+        return self.filename_info["platform_name"]
+
+    @property
+    def orbit_start(self):
+        """Get the orbit number for the start of data."""
+        return int(self["/attr/orbit_start"])
+
+    @property
+    def orbit_end(self):
+        """Get the orbit number for the end of data."""
+        return int(self["/attr/orbit_end"])
+
+    def get_dataset(self, dataset_id, dataset_info):
+        """Get the data."""
+        raise NotImplementedError("This is not implemented in the Base class.")
+
+    def _get_channel_data(self, dataset_id, dataset_info):
+        channel_data = self[dataset_info["file_key"]]
+        channel_data.coords["n_channels"] = MWR_CHANNEL_NAMES
+        channel_data = channel_data.rename({"n_fovs": "x", "n_scans": "y"})
+        return channel_data.sel(n_channels=dataset_id["name"]).drop_vars("n_channels")
+
+
+
+class AWS_EPS_Sterna_MWR_L1BFile(AWS_EPS_Sterna_BaseFileHandler):
+    """Class implementing the AWS/EPS-Sterna MWR L1b Filehandler.
+
+    This class implements the ESA Arctic Weather Satellite (AWS) and EPS-Sterna
+    MWR Level-1b NetCDF reader. It is designed to be used through the
+    :class:`~satpy.Scene` class using the :mod:`~satpy.Scene.load` method with
+    the reader ``"mwr_l1b_nc"``.
```

Comment:
Is this needed?

Reply:
I see what you mean, redundant. As this is nothing unique...
I have removed.